ocp_resources/plan.py

```diff
@@ -59,10 +63,39 @@ def __init__(
         self.network_map_namespace = network_map_namespace
         self.virtual_machines_list = virtual_machines_list
         self.warm_migration = warm_migration
+        self.pre_hook_name = pre_hook_name
+        self.pre_hook_namespace = pre_hook_namespace
+        self.after_hook_name = after_hook_name
+        self.after_hook_namespace = after_hook_namespace
         self.target_namespace = target_namespace or self.namespace
         self.condition_message_ready = self.ConditionMessage.PLAN_READY
         self.condition_message_succeeded = self.ConditionMessage.PLAN_SUCCEEDED
 
+        def get_hook(hook_name, hook_namespace, hook_type):
+            hook = {
+                "hook": {
+                    "name": hook_name,
+                    "namespace": hook_namespace,
+                },
+                "step": hook_type,
+            }
+            return hook
+
+        hooks_array = []
+        if self.pre_hook_name and self.pre_hook_namespace:
+            hooks_array.append(
+                get_hook(self.pre_hook_name, self.pre_hook_namespace, "PreHook")
```

Comment:
Please use are=val when calling a function:

```suggestion
                get_hook(hook_name=self.pre_hook_name, hook_namespace=self.pre_hook_namespace, hook_type="PreHook")
```

Reply:
Done